Imageview/app/src/main/java/com/mycompany/imageview/Search_fragment.java

```diff
@@ -0,0 +1,225 @@
+                    }
+                    else{
+                        link ="http://i.ytimg.com/vi/tntOCGkgt98/maxresdefault.jpg";
+                        Log.i("button_save", "save_first ");
+                    }
+                    db.add_image(new Image_DB(keyword,link));
+                    Log.i("button_save", "save_after ");
+                }
+            });
+        }}
+
+
+    public void transitionToFragment() {
+        Library_fragment library_fragment = new Library_fragment();
+        FragmentManager fm = getActivity().getSupportFragmentManager();
+        FragmentTransaction transaction = fm.beginTransaction();
+        transaction.replace(R.id.container,library_fragment);
+        transaction.commit();
+    }
+
+    public  void create_webview(View view){
+        mWebView = (WebView)view.findViewById(R.id.search_webview);
+        // Enable Javascript
+        WebSettings webSettings = mWebView.getSettings();
+        webSettings.setJavaScriptEnabled(true);
+        webSettings.setBuiltInZoomControls(true);
+        webSettings.setUseWideViewPort(true);
+        webSettings.setLoadWithOverviewMode(true);
+        link ="http://www.canon-europe.com/images/Android-logo_tcm13-1232684.png";
+        mWebView.loadUrl(link);
+    }
+
+    public  void reload_webview(){
+        if (pointer<0){
+            pointer=0;
+        }
+        if (link_array.size()>0) {
+            link = link_array.get(pointer % link_array.size());
+        }
+        mWebView.loadUrl(link);
+    }
+    public void google_search(String search) {
+        // setup requestqueue here. Usually you should set up one queue for global use
+        RequestQueue queue = Volley.newRequestQueue(getActivity());
+
+        // setup the request data
+        search = search.replaceAll(" ","+");
+        String URL = "https://www.googleapis.com/customsearch/v1?";
+        URL+="key=AIzaSyAw-vsCUjYacbHHVDLeSUZPynir7gyGSwE"+"&";
+        URL+="cx=003197243441389598668:b-nzlg3snm8"+"&";
+        URL+="searchType=image&";
+        URL+="q="+search;
+
+        // setup the necessary json body for google search
+        JSONObject body = new JSONObject();
+
+        JsonObjectRequest getRequest = new JsonObjectRequest(
+                Request.Method.GET,
+                URL,
+                body,
+                new Response.Listener<JSONObject>() {
+                    @Override
+                    public void onResponse(JSONObject response) {
+                        // do something with response
+                        parse_json(response);
+                        Log.d("Response", response.toString());
+                    }
+                },
+                new Response.ErrorListener() {
+                    @Override
+                    public void onErrorResponse(VolleyError error) {
+                        Log.e("Error", error.getMessage());
+                    }
+                }
+        );
+
+        queue.add(getRequest);
+    }
```

Comment:
Separating this HTTP request into a separate handler with a callback would be the correct way to do this.